python-avd/pyavd/j2filters/secure_hash.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2023-2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+
+import re
+
+from pyavd._errors import AristaAvdError
+
+try:
+    from passlib.hash import sha512_crypt
+except ImportError as imp_exc:
+    raise AristaAvdError(imp_exc) from imp_exc
+
+HASH_INPUT_TYPE = ["sha512_password"]
+
+
+def _validate_sha512_salt(salt: str | None) -> None:
+    """
+    Validate a given salt value.
+
+    Args:
+        salt: The salt value provided by the user.
+
+    Raises:
+        TypeError: If the salt is not of type 'str'.
+        ValueError: If the salt is greater than 16 characters.
+        ValueError: If the salt contains a character not matching ./0-9A-Za-z
```

Comment:
Lets check if this is base64
```suggestion
        ValueError: If the salt contains a character not matching ./0-9A-Za-z
```

Reply:
it is not - the specification for these valid characters seems to come from mkpasswd in UNIX (which would be what EOS used underneath) - so thats life.

Reply:
ya its slightly different than base64, mainly dot(.) is allowed instead of a plus(+). So a valid salt that contains a dot would fail a base64 check.

I believe mkpasswd uses the C crypt function from my research:
https://man7.org/linux/man-pages/man3/crypt.3.html

Python did implement this as part of its standard library but has since deprecated it since 3.11 - https://docs.python.org/3/library/crypt.html

[passlib](https://pypi.org/project/passlib/) was the best alternative for this that I could find (which I have used)

Reply:
yes for passlib I was just looking for a "nicer" way to refer to this set of allowed characters but it seems there is not a common name

passlib calls it HASH64: https://github.com/glic3rinu/passlib/blob/master/passlib/utils/__init__.py#L1276 but thats not something commonly used it seems so lets keep it this way.